### PR TITLE
API : change begin/end event names

### PR DESCRIPTION
**Pull request by tacaswell**
BeginRunEvent -> RunStart
EndRunEvent -> RunEnd

insert_end_run -> insert_run_end
insert_begin_run -> insert_run_start

sundry variable/arg name changes

### CODE REVIEW

File: conda-recipe/meta.yaml [hunk 1]
```diff
@@ -22,5 +22,12 @@ requirements:
...
-
+    - pyyaml
+
+test:
+  # Python imports
+  imports:
+    - metadataStore
+    - metadataStore.commands
+    - metadataStore.api
+    - metadataStore.conf
```
Comment (ericdill) [inline conda-recipe/meta.yaml]: Is there some reason that `.odm_templates` and `.document` are not included here?

File: metadataStore/commands.py [hunk 2]
```diff
@@ -127,28 +127,28 @@ def insert_begin_run(time, beamline_id, beamline_config=None, owner=None,
     if custom is None:
         custom = {}
 
-    begin_run = BeginRunEvent(time=time, scan_id=scan_id, owner=owner,
+    run_start = RunStart(time=time, scan_id=scan_id, owner=owner,
                               time_as_datetime=__todatetime(time), uid=uid,
                               beamline_id=beamline_id,
                               beamline_config=beamline_config
                               if beamline_config else None,
                               **custom)
```
Comment (ericdill) [inline metadataStore/commands.py]: your whitespace isn't tabbed correctly here

File: metadataStore/commands.py [hunk 3]
```diff
@@ -423,7 +423,7 @@ def find_event_descriptor(begin_run_event):
     """
     event_descriptor_list = list()
     for event_descriptor in EventDescriptor.objects\
```
Comment (ericdill) [inline metadataStore/commands.py]: I thought that `\` made you cringe?

File: metadataStore/commands.py [hunk 4]
```diff
@@ -565,12 +565,12 @@ def find_last(num=1):
...
     Returns
     -------
-    begin_run_event: list
+    run_start: list
         Returns a list of the last ``num`` headers created.
         List of metadataStore.document.Document objects
         **NOTE**: DOES NOT RETURN THE EVENTS.
     """
-    br_objects = [br_obj for br_obj in BeginRunEvent.objects.order_by('-_id')[0:num]]
+    br_objects = [br_obj for br_obj in RunStart.objects.order_by('-_id')[0:num]]
```
Comment (ericdill) [inline metadataStore/commands.py]: not sure you need the `0` portion of the `0:num` here, but it doesn't really matter

File: metadataStore/odm_templates.py [hunk 5]
```diff
@@ -81,13 +81,13 @@ class EndRunEvent(DynamicDocument):
         20 char max.
     """
     time = FloatField(required=True)
-    begin_run_event = ReferenceField(BeginRunEvent, reverse_delete_rule=DENY,
-                                     required=True, db_field='begin_run_id')
+    run_start = ReferenceField(RunStart, reverse_delete_rule=DENY,
+                                     required=True, db_field='run_start_id')
```
Comment (ericdill) [inline metadataStore/odm_templates.py]: whitespace is off here

File: metadataStore/odm_templates.py [hunk 6]
```diff
@@ -120,8 +120,8 @@ class EventDescriptor(DynamicDocument):
 
     Attributes
     ----------
-    begin_run_event : str
-        Globally unique ID to the begin_run document this descriptor is associtaed with.
+    run_start : str
+        Globally unique ID to the run_start document this descriptor is associtaed with.
```
Comment (ericdill) [inline metadataStore/odm_templates.py]: `associtaed` -> `associated`

**Comment by ericdill** [comment]
Nothing major, just some picky PEP8 whitespace stuff

**Comment by ericdill** [comment]
Thanks for the renaming crusade @tacaswell !  I suppose we'll need to have another whenever we do the `metadataStore` -> `metadatastore` one...

File: metadataStore/commands.py (hunk 4 above)
Comment (ghost) [inline metadataStore/commands.py]: You're right. I will change this to conditional find(find_one for limit=1, and find for limit with N). That significantly increases query performance when pulling a single document.

**Comment by ghost** [comment]
Great work!! :+1:  Thanks @tacaswell !!

File: conda-recipe/meta.yaml (hunk 1 above)
Comment (tacaswell) [inline conda-recipe/meta.yaml]: because I wrote this off the top of my head :sheep:

**Comment by tacaswell** [comment]
@ericdill I think I addressed everything

**Pull request merged**